ddtrace/contrib/internal/google_adk/patch.py

```diff
@@ -0,0 +1,256 @@
+                integration.llmobs_set_tags(span, args=args, kwargs=kwargs, response=response_events)
+
+    return _generator()
+
+
+@with_traced_module
+async def _traced_functions_call_tool_async(adk, pin, wrapped, instance, args, kwargs):
+    integration = adk._datadog_integration
+    tool_context = kwargs.get("tool_context", {})
+    # Handle cases where invocation context might not exist (e.g., direct tool calls)
+    agent = None
+    if hasattr(tool_context, "_invocation_context") and hasattr(tool_context._invocation_context, "agent"):
+        agent = tool_context._invocation_context.agent
+    provider_name, model_name = extract_provider_and_model_name(agent=agent)
+    instance = instance or args[0]
+
+    with integration.trace(
+        pin,
+        "%s.%s" % (instance.__class__.__name__, wrapped.__name__),
+        provider=provider_name,
+        model=model_name,
+        kind="tool",
+        submit_to_llmobs=True,
+    ) as span:
+        result = None
+        try:
+            result = await wrapped(*args, **kwargs)
+            return result
+        except Exception as e:
+            span.set_exc_info(type(e), e, e.__traceback__)
+            raise
+        finally:
+            span.set_tag_str("component", "google-adk")
+            kwargs["tool"] = args[0] if args else kwargs.get("tool")
+            kwargs["tool_args"] = args[1] if len(args) > 1 else kwargs.get("args")
+            kwargs["tool_call_id"] = tool_context.function_call_id
+            integration.llmobs_set_tags(
+                span,
+                args=args,
+                kwargs=kwargs,
+                response=result,
+            )
+
+
+@with_traced_module
+async def _traced_functions_call_tool_live(adk, pin, wrapped, instance, args, kwargs):
+    integration = adk._datadog_integration
+    tool_context = kwargs.get("tool_context", {})
+    agent = tool_context._invocation_context.agent
+    provider_name, model_name = extract_provider_and_model_name(agent=agent)
+
+    with integration.trace(
+        pin,
+        "%s.%s" % (instance.__class__.__name__, wrapped.__name__),
+        provider=provider_name,
+        model=model_name,
+        kind="tool",
+        submit_to_llmobs=True,
+    ) as span:
+        result = None
+        try:
+            agen = wrapped(*args, **kwargs)
+            async for item in agen:
+                yield item
+        except Exception as e:
+            span.set_exc_info(type(e), e, e.__traceback__)
+            raise
+        finally:
+            span.set_tag_str("component", "google-adk")
+            kwargs["tool"] = args[0] if args else kwargs.get("tool")
+            kwargs["tool_args"] = args[1] if len(args) > 1 else kwargs.get("args")
+            kwargs["tool_call_id"] = tool_context.function_call_id
+            integration.llmobs_set_tags(
+                span,
+                args=args,
+                kwargs=kwargs,
+                response=result,
+            )
+
+
+@with_traced_module
```

Comment:
let's avoid `@with_traced_module`, the overhead is fairly high, and if we aren't using `Pin` anymore, then it isn't very useful (especially since you already import the module globally)